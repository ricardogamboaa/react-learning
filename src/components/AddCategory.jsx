import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ( { onNewCategory } ) => {
    
    const [inputvalue, setInputvalue] = useState('')
    
    const onInputChange = ( event ) => {
        setInputvalue( event.target.value )
    }
    
    const onSubmit = ( event ) => {
        event.preventDefault();
        const val = inputvalue.trim(); 
        if ( val.length <= 1 ) return;
        onNewCategory( val );
        setInputvalue('');
    }
    
    return (
        <form onSubmit={ onSubmit } aria-label='form' > 
            <input type="text" 
                placeholder='buscar gifs' 
                value={ inputvalue } 
                onChange={ onInputChange } />
        </form>
    )
}
    
    
AddCategory.propTypes = {
    onNewCategory : PropTypes.func.isRequired
}
    