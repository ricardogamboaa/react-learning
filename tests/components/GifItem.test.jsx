import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"



describe('gif item component tests', () => { 
    
    const title = "My Title"
    const url = "https://media0.giphy.com/media/DuVRadBbaX6A8/giphy.gif?cid=071c1690ixpua1kwp4zpbhimm1ir5dkao6rvadzbf2twwg4r&rid=giphy.gif&ct=g";
    
    test('should evaluate against snapshot', () => { 
        
        const { container } = render( <GifItem 
            title={ title }
            url={ url } /> );        
            expect( container ).toMatchSnapshot();
        });
        
        test('should show the image with the url and alt indicated', () => { 
            render(<GifItem title={ title } url={ url } />);
            // screen.debug();

            const { src, alt } = screen.getByRole( 'img' ); 
            expect( src ).toBe( url );
            expect( alt ).toBe( title );
        });

        test('should show title in the component', () => { 
            render(<GifItem title={ title } url={ url } />);
            expect( screen.getByText( title ) );
         });
    });