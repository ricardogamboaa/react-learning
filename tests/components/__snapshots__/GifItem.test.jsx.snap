// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gif item component tests should evaluate against snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="My Title"
      src="https://media0.giphy.com/media/DuVRadBbaX6A8/giphy.gif?cid=071c1690ixpua1kwp4zpbhimm1ir5dkao6rvadzbf2twwg4r&rid=giphy.gif&ct=g"
    />
    <p>
      My Title
    </p>
  </div>
</div>
`;
